### PR TITLE
Create games interfaces

### LINKED ISSUE #5: Create game interfaces
### Describe the feature

Add games interfaces with:
- Entity logic
- Sound logic
- First implementation of types
- Textures and position logic
- Event logic

### Example Feature Purpose

_No response_

### Additional context

This is a fundamental architectural issue, so the provided code will include additional types and code that may not be directly related to the game but are necessary for the creation of the global architecture.

### PR DESCRIPTION
**Pull request by TekMath**
## Changes made

- Added Entity interface
- Added DisplayableEntity interface
- Added SoundProps type
- Added TextureProps type
- Added KeyData type
- Added GameManifest type
- Added Game interface
- Added Vector, UUid types

✨ Make unique_ptr nice to use

### CODE REVIEW

**Review by BxptisteM — DISMISSED** [review]

File: games/entity/IDisplayableEntity.hpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
+/*
+** EPITECH PROJECT, 2024
+** arcade-shared
+** File description:
+** IEntity
```
Comment (Yann-Masson) [inline games/entity/IDisplayableEntity.hpp:5]: This file doesn't name `IEntity` but `IDisplayableEntity`
```suggestion
** IDisplayableEntity
```

File: games/types/GameManifest.hpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/*
+** EPITECH PROJECT, 2024
+** shared-arcade
+** File description:
+** GameManifest
+*/
+
+#pragma once
+#include <string>
```
Comment (Yann-Masson) [inline games/types/GameManifest.hpp:9]: Add a space between `#pragma once` and the `include `
```suggestion
#pragma once

#include <string>
```

File: games/game/IGame.hpp [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+** shared-arcade
+** File description:
+** IGame
+*/
+
+#pragma once
+
+#include <memory>
+
+#include "../types/GameManifest.hpp"
```
Comment (Yann-Masson) [inline games/game/IGame.hpp]:
```suggestion
#include <memory>
#include "../types/GameManifest.hpp"
```

File: games/types/SoundProps.hpp [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+** arcade-shared
+** File description:
+** SoundProps
+*/
+
+#pragma once
+
+#include <string>
+#include <map>
+#include <memory>
```
Comment (Yann-Masson) [inline games/types/SoundProps.hpp]:
```suggestion
#include <map>
#include <string>
#include <memory>
```

File: games/types/TextureProps.hpp [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+** TextureProps
+*/
+
+#pragma once
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include "../../types/types.hpp"
```
Comment (Yann-Masson) [inline games/types/TextureProps.hpp]:
```suggestion
#include <string>
#include "../../types/types.hpp"
```

File: types/types.hpp
Comment (Yann-Masson) [inline types/types.hpp:1]: If you want

**Review by Yann-Masson — CHANGES_REQUESTED** [review]
I have notice that some files use 2 spaces and some others 4 spaces as tabulation. Please update those files and make a choice : 2 spaces or 4 spaces.
Otherwise, it's a really great job! I'm proud to work with you!

**Review by sdragos1 — DISMISSED** [review]

**Review by AxelF44 — DISMISSED** [review]

**Review by Yann-Masson — APPROVED** [review]
Dear Flavien and Mathéo,

I wanted to take a moment to extend my heartfelt congratulations and sincere appreciation for the outstanding contributions both of you have made to our project, especially in the recent pull request.

Flavien, your dedication to detail and your ability to tackle complex challenges head-on have been truly impressive. Your expertise and insights have consistently elevated the quality of our project, and your recent contributions in the pull request have once again highlighted your commitment to excellence.

Mathéo, your innovative thinking and proactive approach to problem-solving have been invaluable to our team. Your willingness to explore new ideas and your exceptional coding skills have greatly enhanced our project's capabilities. Your recent work in the pull request has been nothing short of exceptional, and it's clear that your passion for our project shines through in everything you do.

Together, your collaborative efforts have fostered a dynamic and supportive team environment. Your willingness to share knowledge, provide feedback, and support each other has been instrumental to our success. It's inspiring to see how your teamwork has propelled us forward, and I have no doubt that our project is stronger because of your combined efforts.

On behalf of the entire team, I want to express my deepest gratitude for your hard work, dedication, and unwavering commitment. Your contributions have not only helped us achieve our goals but have also set a high standard for excellence within our team.

Congratulations, Flavien and Mathéo, on a job well done! I look forward to continuing our journey together and achieving even greater milestones in the future.

Warm regards ✈,

**Review by AxelF44 — APPROVED** [review]

**Review by sdragos1 — APPROVED** [review]

**Pull request merged**